src/pfe/file-watcher/server/src/projects/OdoExtensionProject.ts

```diff
@@ -0,0 +1,275 @@
+
+/**
+ * @interface
+ * @description Interface for odo extension project configuration.
+ */
+interface OdoExtensionProjectConfig {
+    requiredFiles: string[];
+    buildLogs: string[];
+    appLogs: string[];
+    capabilities?: ProjectCapabilities;
+}
+
+/**
+ * @class
+ * @extends IExtensionProject
+ * @description The class representing a project handler for odo extension projects. It is
+ * used for storing odo extension metadata and calling odo extension operations (eg. create, update, remove, etc).
+ */
+export class OdoExtensionProject implements IExtensionProject {
+
+    supportedType: string;
+
+    private fullPath: string;
+    private config: OdoExtensionProjectConfig;
+
+    /**
+     * @constructor
+     * @description Constructor.
+     *
+     * @param type <Required | String> - Set the supported type.
+     */
+    constructor(type: string) {
+        this.supportedType = type;
+    }
+
+    /**
+     * @function
+     * @description Initialize this extension project.
+     *
+     * @param projectInfo <Required | ProjectInfo> - The metadata information for the project.
+     */
+    init = async (projectInfo: ProjectInfo): Promise<void> => {
+        this.fullPath = projectInfo.extensionID;
+        this.config = await fs.readJson(path.join(this.fullPath, ".odo-extension-config.json"));
+    }
+
+    /**
+     * @function
+     * @description Create operation for the project.
+     *
+     * @param operation <Required | Operation> - The create operation.
+     */
+    create = (operation: Operation): void => {
+        projectUtil.containerCreate(operation, path.join(this.fullPath, "odo-extension-entrypoint.sh"), "create");
+    }
+
+    /**
+     * @function
+     * @description Update operation for the project.
+     *
+     * @param operation <Required | Operation> - The update operation.
+     * @param changedFiles <Optional | projectEventsController.IFileChangeEvent[]> - The file changed event array.
+     */
+    update = (operation: Operation, changedFiles?: projectEventsController.IFileChangeEvent[]): void => {
+        projectUtil.containerUpdate(operation, path.join(this.fullPath, "odo-extension-entrypoint.sh"), "update");
+    }
+
+    /**
+     * @function
+     * @description Rebuild operation for the project.
+     *
+     * @param projectInfo <Required | ProjectInfo> - The metadata information for the project.
+     */
+    rebuild = async (projectInfo: ProjectInfo): Promise<void> => {
+        await projectUtil.runScript(projectInfo, path.join(this.fullPath, "odo-extension-entrypoint.sh"), "rebuild");
+    }
+
+    /**
+     * @function
+     * @description Delete operation for the project.
+     *
+     * @param projectInfo <Required | ProjectInfo> - The metadata information for the project.
+     */
+    deleteContainer = async (projectInfo: ProjectInfo): Promise<void> => {
+        await projectUtil.containerDelete(projectInfo, path.join(this.fullPath, "odo-extension-entrypoint.sh"));
```

Comment:
My guess is this code is trying to delet the entrypoint.sh file during the container deletion operation. Why do we delete the entrypoint.sh during a containerDelete?

Reply:
We don't delete the entrypoint.sh, we just call the entrypoint.sh to do the container deletion operation, it's same as create and update cases, but for delete case we call entrypoint.sh with `remove` parameter here: https://github.com/eclipse/codewind/blob/master/src/pfe/file-watcher/server/src/projects/projectUtil.ts#L624